packages/react-components/src/components/ChatMessage/ChatMessageContent.tsx

```diff
@@ -76,21 +76,27 @@ const MessageContentWithLiveAria = (props: MessageContentWithLiveAriaProps): JSX
 };
 
 const MessageContentAsRichTextHTML = (props: ChatMessageContentProps): JSX.Element => {
+  const {
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+    // message is used only in useEffect that is under teams-inline-images-and-file-sharing cc
+    message,
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+    attachmentsMap,
+    /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
+    onFetchAttachment
+  } = props;
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
   useEffect(() => {
-    const attachments = props.message.attachedFilesMetadata?.filter((fileMetadata) => {
-      return fileMetadata.attachmentType === 'inlineImage';
+    if (!attachmentsMap || !onFetchAttachment) {
+      return;
+    }
+    const attachments = message.attachedFilesMetadata?.filter((fileMetadata) => {
+      return fileMetadata.attachmentType === 'inlineImage' && attachmentsMap[fileMetadata.id] === undefined;
     });
-
-    if (props.attachmentsMap && attachments) {
-      attachments.forEach((fileMetadata) => {
-        if (props.onFetchAttachment && props.attachmentsMap && props.attachmentsMap[fileMetadata.id] === undefined) {
-          props.onFetchAttachment([fileMetadata], props.message.messageId);
-          return;
-        }
-      });
+    if (attachments && attachments.length > 0) {
+      onFetchAttachment(attachments, message.messageId);
```

Comment:
I kind of remember @JoshuaLai mentioned that there's a reason to call onFetchAttachment on each attachment?

Reply:
This is fine, we had it fetching multiple before because our adapter couldn't handle using the full attachment list, but after my refactor a little while back we can support this type of functionality now